src/uct/ib/Makefile.am

```diff
@@ -11,7 +11,8 @@ module_LTLIBRARIES    = libuct_ib.la
 libuct_ib_la_CPPFLAGS = $(BASE_CPPFLAGS) $(IBVERBS_CPPFLAGS)
 libuct_ib_la_CFLAGS   = $(BASE_CFLAGS)
 libuct_ib_la_LIBADD   = $(top_builddir)/src/ucs/libucs.la \
-                        $(top_builddir)/src/uct/libuct.la
+                        $(top_builddir)/src/uct/libuct.la \
+                        $(top_builddir)/src/uct/cuda/libuct_cuda.la
```

Comment:
this will break UCT module separation

Reply:
Without the additional UCT changes to be able to query another UCT component's distance, I don't see how we can break the dependence. @yosefe Is there a quick fix for this?

Reply:
i was thinking that maybe add a cuda submodule for uct_ib like we do for perftest and ucm
but this is not straight forward.
i think it would be better to do it in UCP and should not be much more work
@Akshay-Venkatesh can we put this to v1.6.1 (which could happen pretty soon after v1.6.0), or you consider this a blocker for v1.6.0?

Reply:
Addressing in UCP sounds cleaner. 

1.6.1 is ok for me. If we handle this in UCP, then adding this in master is much better but I guess 1.6.1 will be additions only on top of 1.6.0. Is that right?

Reply:
yes. i suggest we fix it in master by UCP method, then add it on top of v1.6.0 to create v1.6.1